### PR TITLE
feat(ramps): update isNativeTokenBuyable to include BTC 

### LINKED ISSUE #25657: [Bug]: "Buy ETH" card appears when user has > 0 balance
### Describe the bug

When the fiat conversion request fails, the "Buy ETH to get started" card appears despite having sufficient ETH in the wallet. 

### Expected behavior

The Ramps Card should not appear if the user has native tokens in their wallet

### Screenshots/Recordings

Screenshot 2024-06-11 at 09 58 31

### Steps to reproduce

this bug happened intermittently when the request to convert crypto to fiat fails. It is not easy to reproduce.

### Error messages or log output

_No response_

### Version

11.16.9

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by georgeweiler**
## **Description**

- updates the `isNativeTokenBuyable` function to consider bitcoin
- updates the zero balance state for the ramps card to look at token balance instead of fiat balance

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25621?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25657

## **Manual testing steps**
```
# METAMASK_RAMP_API_CONTENT_BASE_URL=https://on-ramp-content.api.cx.metamask.io/ -- BTC DISABLED
# METAMASK_RAMP_API_CONTENT_BASE_URL=https://on-ramp-content.uat-api.cx.metamask.io/ -- BTC ENABLED
```

The following QA steps will allow you to test the permutations of this feature:

**Verify that ERC20 Ramps card is still working for 0 balance accounts**
1. Install the Metamask flask extension
2. Verify that the ERC20 Ramps card is still working for the selected network on accounts that have zero token balance

**Verify ERC20 Ramps card is hidden for accounts with > 0 token balance**
1. Verify that the ERC20 Ramps card is does not appear on accounts that have > 0 token balance

**Verify that BTC Ramps card is working**
1. configure `.metamaskrc` to access ramps UAT
2. Install the Metamask flask extension and create a brand new wallet
3. Add a bitcoin account via ramps-dev portfolio environment. It should have 0 BTC.
4. Verify that the BTC Ramps card appears
5. Verify that the correct URL params are added when clicking the "buy" CTA (deeplink UX does not work yet)

**Verify that BTC Ramps card is hidden for accounts with > 0 BTC**
1. configure `.metamaskrc` to access ramps UAT
2. Install the Metamask flask extension and recover a wallet that has > 0 BTC balance
3. Add the bitcoin account via ramps-dev portfolio environment
4. Verify that the BTC Ramps card does not appear

**Verify that BTC Ramps card is hidden in PROD for all accounts**
1. configure `.metamaskrc` to access ramps PROD
2. Install the Metamask flask extension create a new wallet
3. Add a bitcoin account via ramps-dev portfolio environment. It should have 0 BTC
4. Verify that the BTC Ramps card does not appear

## **Screenshots/Recordings**

BTC card appears for 0 balance accounts:
Screenshot 2024-07-01 at 6 33 30 PM

BTC card does not appear for > 0 balance:
Screenshot 2024-07-09 at 10 14 07 AM

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/asset-list/asset-list.js [hunk 1]
```diff
@@ -98,24 +100,31 @@ const AssetList = ({ onClickAsset, showTokensLinks }) => {
...
   const shouldShowBuy = isBuyableChain && balanceIsZero;
   ///: END:ONLY_INCLUDE_IF
 
   const isEvm = useSelector(getMultichainIsEvm);
-
   // NOTE: Since we can parametrize it now, we keep the original behavior
   // for EVM assets
   const shouldShowTokensLinks = showTokensLinks ?? isEvm;
 
+  const isBTC = useSelector(getMultichainIsBitcoin);
```
Comment (ccharly) [inline ui/components/app/asset-list/asset-list.js:126]: I would go with `isBtc` to match the convention with use elsewhere (see `isEvm`):

```suggestion
  const isBtc = useSelector(getMultichainIsBitcoin);
```

File: ui/components/app/asset-list/asset-list.js [hunk 2]
```diff
@@ -133,7 +142,13 @@ const AssetList = ({ onClickAsset, showTokensLinks }) => {
       {
         ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
         shouldShowBuy ? (
-          <RampsCard variant={RAMPS_CARD_VARIANT_TYPES.TOKEN} />
+          <RampsCard
+            variant={
+              isBTC
```
Comment (ccharly) [inline ui/components/app/asset-list/asset-list.js:147]:
```suggestion
              isBtc
```

File: ui/components/app/wallet-overview/btc-overview.test.tsx [hunk 3]
```diff
@@ -40,43 +41,63 @@ const mockNonEvmAccount = {
...
-      // (Multichain) RatesController
-      fiatCurrency: 'usd',
-      rates: {
-        [Cryptocurrency.Btc]: {
-          conversionRate: '1.000',
-          conversionDate: 0,
-        },
+const mockBtcChain = {
+  active: true,
+  chainId: 'bip122:000000000019d6689c085ae165831e93',
```
Comment (ccharly) [inline ui/components/app/wallet-overview/btc-overview.test.tsx:46]: Maybe use the constant here, unless this was intended?

```suggestion
  chainId: MultichainNetworks.BITCOIN,
```

File: ui/ducks/ramps/types.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export type AggregatorNetwork = {
   active: boolean;
-  chainId: number;
+  chainId: number | string;
```
Comment (ccharly) [inline ui/ducks/ramps/types.ts:3]: Question: Did you add a `string` for CAIP-2 identifiers? If yes, there's a `CaipChainId` type in `@metamask/utils`

File: ui/selectors/multichain.ts [hunk 5]
```diff
@@ -315,6 +327,15 @@ export function getMultichainSelectedAccountCachedBalance(
     : getBtcCachedBalance(state);
 }
 
+export const getMultichainSelectedAccountCachedBalanceIsZero = createSelector(
+  [getMultichainIsEvm, getMultichainSelectedAccountCachedBalance],
+  (isEVM, balance) => {
+    const base = isEVM ? 16 : 10;
```
Comment (ccharly) [inline ui/selectors/multichain.ts:333]:
```suggestion
  (isEvm, balance) => {
    const base = isEvm ? 16 : 10;
```

**Review by ccharly — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]
lgtm

**Comment by gantunesr** [comment]
Hey @georgeweiler, can you add the test step and also some screenshots to the PR for documentation purposes?

**Comment by bkirb** [comment]
LGTM for QA ✅

**Pull request merged**